### DISCUSSION TITLE
v0.5.1 Release

### DISCUSSION THREAD

**Discussion by Peefy** [kcl-lang/kcl#624] [Announcements]
## What's Changed
* feat: add position for each name in identifier by @He1pa in https://github.com/kcl-lang/kcl/pull/608
* chore: add the maintainer file by @Peefy in https://github.com/kcl-lang/kcl/pull/609
* Refactor: refactor find def in LSP by @He1pa in https://github.com/kcl-lang/kcl/pull/606
* chore: update maintainers and adopters. by @Peefy in https://github.com/kcl-lang/kcl/pull/611
* bugfix: if_stmt orelse postion by @He1pa in https://github.com/kcl-lang/kcl/pull/612
* refactor: enhance kcl and its tool cli error message and add more tests. by @Peefy in https://github.com/kcl-lang/kcl/pull/618
* ci: add backport action by @Peefy in https://github.com/kcl-lang/kcl/pull/617
* fix: fix performance regression on resolver::parse_doc_string by @He1pa in https://github.com/kcl-lang/kcl/pull/619
* feat: lsp completion of pkg, var and schema attr by @He1pa in https://github.com/kcl-lang/kcl/pull/614
* chore: bump serde_yaml to latest to solve security issues and error test cases by @Peefy in https://github.com/kcl-lang/kcl/pull/623
* feat: support for taking mod relative paths as input. by @zong-zhe in https://github.com/kcl-lang/kcl/pull/622

**Full Changelog**: https://github.com/kcl-lang/kcl/compare/v0.5.0...v0.5.1

<hr /><em>This discussion was created from the release <a href='https://github.com/kcl-lang/kcl/releases/tag/v0.5.1'>v0.5.1 Release</a>.</em>

**Comment by nkabir** [comment]
I would like to use `kcl-py` but am not able to find documentation indicating that is stays in sync with new releases. Will code written with `kcl-py` work the same as the `kcl` command line tool with each new release?

**Reply by Peefy** [reply]
Hi @nkabir. Do you mean the [KCL python SDK](https://github.com/kcl-lang/kcl-py)? By the way, what are the specific scenarios in which you use KCL? Are you willing to fill out this survey to help us better improve KCL: https://github.com/kcl-lang/kcl/issues/578

**Reply by nkabir** [reply]
Yes. I was referring to the KCL Python SDK.

I am using KCL everywhere. It's a versatile tool that helps structure configuration and application data. I'll fill out the survey.

**Reply by nkabir** [reply]
Just following up--does the KCL Python SDK track the KCL releases? Or is it better to use the KCL command line tool to gain access to the latest features and bug fixes?

**Reply by Peefy** [reply]
> Just following up--does the KCL Python SDK track the KCL releases?

Yes, we will track the release v0.5.0 - v0.5.2: https://github.com/kcl-lang/kcl-py/issues/23
Besides, we suggest using the KCL command line to verify the latest features first. If there is an update to KCL Python SDK, I will ping you. ❤️
